### PR TITLE
CAL-258 -PORT- Extract NitfPreStoragePlugin into its own bundle/feature

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Port from 0.2.x https://github.com/codice/alliance/pull/259
Extract NitfPreStoragePlugin into its own bundle/feature. This enables turning off the thumbnail generation for NITF files.
This is provides a way to manage OOM issues encountered with ingesting large nitf files. The true fix for the issue will be the async processing framework that is still under development.
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@kcwire
@pklinef

#### How should this be tested?
Verify build passes
#### What are the relevant tickets?

[CAL-258](https://codice.atlassian.net/browse/CAL-258)
#### Checklist:
- [X] Documentation Updated
	- [ ] Change Log Updated
- [X] Update / Add Unit Tests
- [X] Update / Add Integration Tests

### CODE REVIEW

File: catalog/imaging/imaging-plugin-nitf/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 1]
```diff
@@ -14,17 +14,29 @@
...
+    <OCD name="Nitf Render Plugin"
+         id="NITF_Render_Plugin">
         <AD
                 description="Maximum length of longest side of NITF overview image in pixels. The input transformer will calculate the size of the shorter side so that the overview will have the same aspect ratio as the original."
                 name="Overview image maximum side length (pixels)" id="maxSideLength" required="true"
                 type="Integer" default="1024"/>
+        <AD
+                description="Maximum NITF file size in Megabytes to create thumbnails/overviews for. Rendering large nitfs can cause out-of-memory exceptions."
+                name="Maximum NITF file size to process in Megabytes" id="maxNitfSizeMB" required="true"
+                type="Integer" default="10"/>
```
Comment (rzwiefel) [inline catalog/imaging/imaging-plugin-nitf/src/main/resources/OSGI-INF/metatype/metatype.xml]: is a default of 10MB too small?

Comment (rzwiefel) [inline catalog/imaging/imaging-plugin-nitf/src/main/resources/OSGI-INF/metatype/metatype.xml]: one of the two nitfs I commonly use is 15MB, And i was under the empression this many nitfs we use are on the small side of realistic.

Comment (clockard) [inline catalog/imaging/imaging-plugin-nitf/src/main/resources/OSGI-INF/metatype/metatype.xml]: The reported issue was with nitf files that were 14MB. Things to consider when picking this size are:
- Each NITF will take up ~4x the file size in memory
- Must account for up to 8 threads simultaneously ingesting (directory monitoring allows up to 8 parallel ingest threads)
- The current heap space for the whole system, default is 2Gig

10MB I picked a small value but was looking for feedback/recommendations. From my testing I don't think we would want to do anything more that 30MB as a default.

Comment (pklinef) [inline catalog/imaging/imaging-plugin-nitf/src/main/resources/OSGI-INF/metatype/metatype.xml]: 2Gig seems very low for a streaming FMV or NITF rendering system.  I wonder if we need better defaults for Alliance and then adjust limits like this one to reflect that change.

@jlcsmith @kcwire Thoughts?

Comment (pklinef) [inline catalog/imaging/imaging-plugin-nitf/src/main/resources/OSGI-INF/metatype/metatype.xml]: Not for this PR but should we also consider off heap or on disk options?

Comment (clockard) [inline catalog/imaging/imaging-plugin-nitf/src/main/resources/OSGI-INF/metatype/metatype.xml]: If we can change the jvm heap to 10Gigs then we could support a default to 400MB and 5 ingest threads. We would also need to change the heap size on downstream projects as well.

**Pull request merged**